browser/lib/util/webstorage.js

```diff
@@ -9,25 +10,26 @@ var WebStorage = (function() {
 	 * somewhat roundabout way. (If unsupported or no window object,
 	 * will throw on accessing a property of undefined) */
 	try {
-		window.sessionStorage.setItem(test, test);
-		window.sessionStorage.removeItem(test);
+		globalObject.sessionStorage.setItem(test, test);
+        globalObject.sessionStorage.removeItem(test);
 		sessionSupported = true;
 	} catch(e) {
 		sessionSupported = false;
 	}
 
 	try {
-		window.localStorage.setItem(test, test);
-		window.localStorage.removeItem(test);
+        globalObject.localStorage.setItem(test, test);
+        globalObject.localStorage.removeItem(test);
 		localSupported = true;
 	} catch(e) {
 		localSupported = false;
 	}
 
-	function WebStorage() {}
+
+    function WebStorage() {}
 
 	function storageInterface(session) {
-		return session ? window.sessionStorage : window.localStorage;
+		return session ? globalObject.sessionStorage : globalObject.localStorage;
```

Comment:
Thanks. I get that `window` won't be defined, but if we just wanted to not support webstorage for nativescript, we could just put `if(typeof window == 'undefined') return;` at the top rather than bother having logic to make it look in the global object which won't have the required properties anyway. (Unless someone installed your plugin, which I don't think is something we should bet on). (Node has `WebStorage` undefined too, so other parts of the library expect that).

Re how webstorage is used: sessionstorage is used to persist connection state across page refreshes; localstorage is used to store transport preferences.

Questions:
- I'm guessing persisting connection state across page refreshes isn't a thing in nativescript, and once the client is initialized, it will exist for the duration that the app is open, so sessionstorage won't be useful. Is that right?
- But nativescript does support both xhr and websocket, so persisting transport preferences in localstorage is useful - right?

If that's the case, then perhaps we should have `sessionsSupported` be false, but `localSupported` be true and `WebStorage.get/set/remove` implemented here as a layer around the nativescript `application-settings` mechanism. What do you think?

For cleanness, if we're doing that, then should probably implement a separate `webstorage.js` for nativescript and include that separately in the grunt concatenator (ie split it out of `ablyFiles`).